clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -271,7 +276,7 @@ private long sendProducerData(long now) {
             }
         }
 
-        List<ProducerBatch> expiredBatches = this.accumulator.expiredBatches(this.requestTimeout, now);
+        List<ProducerBatch> expiredBatches = this.accumulator.expiredBatches(this.deliveryTimeoutMs, now);
```

Comment:
It seems an existing issue. When we expire the batches here. The memory of those batches will be deallocated. It seems that we will deallocate the same batch again when the ProduceResponse returns?